### DISCUSSION TITLE
Headless CMS API

### DISCUSSION THREAD

**Discussion by hubyhuby** [GetPublii/Publii#1540] [Ideas]
Hi there
Is it possible to use Publii in headless mode ?
What we want to achieve is using Ruby on rails to generate  specific pages (lisitngs) for Publii

**Comment by H-Hp** [comment]
Publii is a static site generator designed primarily for creating and managing websites through a user-friendly graphical interface. However, it does not have a built-in headless mode or an API for generating content programmatically.
Publii is built on top of Electron, which is a framework for building cross-platform desktop applications using web technologies like HTML, CSS, and JavaScript. The application itself does not expose an API or a command-line interface for generating content programmatically.
That being said, it is possible to use Publii in combination with other tools and frameworks to achieve your goal of generating specific pages (listings) for Publii using Ruby on Rails. Here's a high-level approach you could consider:

Use Publii to set up your site and create templates: In Publii, you can design your site's structure, create templates for different types of pages (e.g., listings), and define the necessary data fields.
Export your Publii site as a ZIP file: Publii allows you to export your entire site as a ZIP file, which includes all the templates, assets, and configuration files.
Use Ruby on Rails to generate content and build the site: In your Ruby on Rails application, you can write code to generate the necessary data (e.g., listings) and build the site using the exported Publii templates and configuration. You can leverage Ruby gems like nokogiri to parse and manipulate the HTML templates, and zip to create a new ZIP file containing the generated site.
Import the generated site back into Publii: Once you've generated the site using your Ruby on Rails application, you can import the resulting ZIP file back into Publii. This will allow you to preview the site and make any additional changes or tweaks using Publii's graphical interface.

While this approach is not a direct integration with Publii's headless mode (since it doesn't have one), it allows you to leverage the strengths of both Publii (for site setup and templating) and Ruby on Rails (for generating dynamic content and building the site programmatically).
Keep in mind that this approach may require some additional effort to set up and maintain, as you'll need to ensure that your Ruby on Rails code correctly interprets and generates the site according to Publii's template structure and configuration.
